src/app/domain/authentication/models/User.ts

```diff
@@ -0,0 +1,137 @@
+import CozyClient, { Q } from 'cozy-client'
+import { ClientCapabilities } from 'cozy-client/types/types'
+import flag from 'cozy-flags'
+
+import { devlog } from '/core/tools/env'
+
+export const canAuthWithOIDC = (client: CozyClient): boolean | undefined => {
+  return isOidcAuth(client)
+}
+
+export const shouldCreatePassword = async (
+  client: CozyClient
+): Promise<boolean> => {
+  const passwordDefined = await isPasswordDefined(client)
+
+  return (
+    !(await IsVaultInstalled(client)) &&
+    (isOidcAuth(client) || isMagicLinkAuth(client)) &&
+    !passwordDefined
+  )
+}
+
+const IsVaultInstalled = async (client: CozyClient): Promise<boolean> => {
+  try {
+    const vault = await client
+      .getStackClient()
+      .fetchJSON<{ extension_installed?: boolean }>(
+        'GET',
+        '/data/io.cozy.settings/io.cozy.settings.bitwarden',
```

Comment:
Here we can client.query `Q('io.cozy.settings').getById('io.cozy.settings.bitwarden')` instead

Reply:
You can find an exemple here: https://github.com/cozy/cozy-flagship-app/blob/3381ab6ffb4c4bd80d2a00e983f21fd4e98a82d1/src/libs/functions/openApp.js#L59-L67

Reply:
- [ ] Updated